adapters/amqp-vertx-base/src/main/java/org/eclipse/hono/adapter/amqp/VertxBasedAmqpProtocolAdapter.java

```diff
@@ -754,6 +765,17 @@ protected void handleRemoteSenderOpenForCommands(
         });
     }
 
+    private Future<Boolean> isTerminalError(final Throwable error, final AmqpContext ctx,
+            final SpanContext spanContext) {
+        final ResourceIdentifier address = ctx.getAddress();
+        if (address == null) {
+            return Future.succeededFuture(false);
+        } else {
+            return isTerminalError(error, address.getTenantId(), address.getResourceId(), ctx.getAuthenticatedDevice(),
```

Comment:
I think you can't use `ctx.getAddress().getTenantId()` here.
The address is the original address from the request, so it could just be `event`.

When the event message request gets processed, a separate ResourceIdentifier with included tenant and deviceId gets created in `AbstractProtocolAdapterBase.validateAddress()` and that one is used then, see
https://github.com/eclipse/hono/blob/f227638f19988d034153812d9a462a50e10c152e/adapters/amqp-vertx-base/src/main/java/org/eclipse/hono/adapter/amqp/VertxBasedAmqpProtocolAdapter.java#L1119-L1130